src/main.js

```diff
@@ -98,28 +98,111 @@ document.addEventListener('DOMContentLoaded', function () {
       const isHtml = /<[a-z][\s\S]*>/i.test(newText);
 
       if (!didToggleFullScreen) {
-        $('#summernote').summernote('fullscreen.toggle');
+        summernote.summernote('fullscreen.toggle');
         didToggleFullScreen = true;
       }
 
       if (newNoteLoad && !isHtml) {
         newText = textToHTML(newText);
       }
 
+      let renderNote = false;
+      const isUnsafeContent = checkIfUnsafeContent(newText);
+
+      if (isUnsafeContent) {
+        const currentNotePreferences = getCurrentNotePreferences();
+        if (!currentNotePreferences) {
+          showUnsafeContentAlert().then((result) => {
+            if (result) {
+              setNotePreferences('renderNote', result);
```

Comment:
I think this key should be called something more descriptive, like `trustUnsafeContent`